### PR TITLE
docs: Fix minor issues in docs/usage/applications.rst

### LINKED ISSUE #2688: Docs(bug): The `on_startup` / `on_shutdown` links are stale
### Summary

They no longer link anywhere.

https://docs.litestar.dev/latest/usage/applications.html#using-application-state

image

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh Litestar dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/2688">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2688/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### LINKED ISSUE #3110: Docs: Minor inconsistencies in description and example for "Applications" chapter
### Summary

These are small inconsistencies I spotted while reading this chapter. 

Using `state` as kwarg for `on_startup` and `on_shutdown`:
- Link: https://docs.litestar.dev/latest/usage/applications.html#using-application-state
- Paragraph: "As seen in the examples for the [on_startup](https://docs.litestar.dev/latest/usage/applications.html#before-after-startup) / [on_shutdown](https://docs.litestar.dev/latest/usage/applications.html#before-after-shutdown) , callables passed to these hooks can receive an optional kwarg called state, which is the application’s state object."
- Problem: The examples at the top of the chapter do not use the `state` kwargs argument. Instead they use `app` and then fetch the state from that object .
- Suggested solution: Since use of `state` as reserved kwarg is documented [elsewhere](https://docs.litestar.dev/latest/usage/routing/handlers.html#reserved-keyword-arguments), I assume the example file just needs an update.

The `after_exception` hook:
- Link: https://docs.litestar.dev/latest/usage/applications.html#after-exception
- Paragraph: "The after_exception hook takes a [sync or async callable](https://docs.litestar.dev/latest/reference/types.html#litestar.types.AfterExceptionHookHandler) that is called with three arguments: the exception that occurred, the ASGI scope of the request or websocket connection, and the application state."
- Problem: Documentation states that 3 arguments need to be passed, example only uses 2 arguments: `exc` and `scope`
- Suggested solution: Again, I assume the example is just outdated.

The `before_send` hook:
- Link: https://docs.litestar.dev/latest/usage/applications.html#before-send
- Paragraph: "The hook receives the message instance and the application state."
- Problem: The example receives scope instead of state
- Suggested solution: Not sure if this is even a problem or just fuzzy wording. But since the sections before referred to the kwarg `state` I think this should be made more consistent by either using `state` instead of `scope` in the example or to update the description to use `scope` instead of `state`.

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3110">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3110/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by aranvir**
## Description

Suggestions for better alignment between descriptions and code examples. Also attempted to fix stale link (but not used to rst, so not sure if that works like that).

## Closes

Closes #3110
Closes #2688

### CODE REVIEW

File: docs/usage/applications.rst [hunk 1]
```diff
@@ -93,8 +94,9 @@ On the other hand, the shutdown hooks are invoked in their specified order.
 Using Application State
 -----------------------
 
-As seen in the examples for the `on_startup <#before-after-startup>`_ / `on_shutdown <#before-after-shutdown>`_ ,
-callables passed to these hooks can receive an optional kwarg called ``state``, which is the application's state object.
+As seen in the examples for the `on_startup / on_shutdown <#startup-and-shutdown>`_ ,
+callables passed to these hooks can receive an optional kwarg called ``app``, through which is the application's state object
```
Comment (provinzkraut) [inline docs/usage/applications.rst:98]:
```suggestion
callables passed to these hooks can receive an optional kwarg called ``app``, through which the application's state object
```

**Review by provinzkraut — CHANGES_REQUESTED** [review]

File: docs/usage/applications.rst [hunk 2]
```diff
@@ -93,8 +94,9 @@ On the other hand, the shutdown hooks are invoked in their specified order.
...
+and other properites can be accessed.
```
Comment (provinzkraut) [inline docs/usage/applications.rst:99]:
```suggestion
callables passed to these hooks can receive an optional kwarg called ``app``, through which the application's state object
and other properties can be accessed.
```

**Review by provinzkraut — CHANGES_REQUESTED** [review]

File: docs/usage/applications.rst [hunk 3]
```diff
@@ -25,6 +25,7 @@ and Route Handlers should be registered on it.
 
     * :ref:`Routing - Registering Routes <usage/routing/overview:Registering Routes>`
 
+.. _startup-and-shutdown:
```
Comment (JacobCoffee) [inline docs/usage/applications.rst:28]: I thought that with [`autosectionlabel`](https://www.sphinx-doc.org/en/master/usage/extensions/autosectionlabel.html#module-sphinx.ext.autosectionlabel) we didn't need to do things like this

File: docs/usage/applications.rst [hunk 4]
```diff
@@ -93,8 +94,9 @@ On the other hand, the shutdown hooks are invoked in their specified order.
 Using Application State
 -----------------------
 
-As seen in the examples for the `on_startup <#before-after-startup>`_ / `on_shutdown <#before-after-shutdown>`_ ,
-callables passed to these hooks can receive an optional kwarg called ``state``, which is the application's state object.
+As seen in the examples for the `on_startup / on_shutdown <#startup-and-shutdown>`_ ,
```
Comment (JacobCoffee) [inline docs/usage/applications.rst:97]:
```suggestion
As seen in the examples for the `on_startup / on_shutdown <#startup-and-shutdown>`_,
```

File: docs/usage/applications.rst (hunk 3 above)
Comment (aranvir) [reply docs/usage/applications.rst]: Updated to use section reference.

Only did a quick research on rst yesterday and didnt see that as an option, but it's cleaner.

Comment (aranvir) [reply docs/usage/applications.rst]: That didnt work out. Does it need a full path? Like `<usage/application:Startup and Shutdown>`?

Comment (provinzkraut) [reply docs/usage/applications.rst]: You just need to get rid of the `:ref:` role, then it should work.

**Review by provinzkraut — APPROVED** [review]

**Comment by provinzkraut** [comment]
@all-contributors add @aranvir for docs

**Pull request merged**